aws-rds-optiongroup/src/main/java/software/amazon/rds/optiongroup/Translator.java

```diff
@@ -0,0 +1,220 @@
+                .optionsToInclude(optionsToIncludeSdk)
+                .optionsToRemove(optionNamesToRemove)
+                .applyImmediately(Boolean.TRUE)
+                .build();
+    }
+
+    public static DeleteOptionGroupRequest deleteOptionGroupRequest(final ResourceModel model) {
+        return DeleteOptionGroupRequest.builder()
+                .optionGroupName(model.getOptionGroupName())
+                .build();
+    }
+
+    static Set<OptionConfiguration> translateOptionConfigurationsFromSdk(final Collection<software.amazon.awssdk.services.rds.model.Option> options) {
+        return Optional.ofNullable(options).orElse(Collections.emptyList())
+                .stream()
+                .map(option -> {
+                    return OptionConfiguration.builder()
+                            .dBSecurityGroupMemberships(translateDBSecurityMembershipsFromSdk(option.dbSecurityGroupMemberships()))
+                            .optionName(option.optionName())
+                            .optionSettings(translateOptionSettingsFromSdk(option.optionSettings()))
+                            .optionVersion(option.optionVersion())
+                            .port(option.port())
+                            .vpcSecurityGroupMemberships(translateVpcSecurityGroupMembershipsFromSdk(option.vpcSecurityGroupMemberships()))
+                            .build();
+                })
+                .collect(Collectors.toSet());
+    }
+
+    static List<software.amazon.awssdk.services.rds.model.OptionConfiguration> translateOptionConfigurationsToSdk(final Collection<OptionConfiguration> options) {
+        return Optional.ofNullable(options).orElse(Collections.emptyList())
+                .stream()
+                .map(option -> {
+                    return software.amazon.awssdk.services.rds.model.OptionConfiguration
+                            .builder()
+                            .optionName(option.getOptionName())
+                            .dbSecurityGroupMemberships(translateDBSecurityMembershipsToSdk(option.getDBSecurityGroupMemberships()))
+                            .optionSettings(translateOptionSettingsToSdk(option.getOptionSettings()))
+                            .optionVersion(option.getOptionVersion())
+                            .port(option.getPort())
+                            .vpcSecurityGroupMemberships(translateVpcSecurityGroupMembershipsToSdk(option.getVpcSecurityGroupMemberships()))
+                            .build();
+                })
+                .collect(Collectors.toList());
+    }
+
+    static Set<OptionSetting> translateOptionSettingsFromSdk(final Collection<software.amazon.awssdk.services.rds.model.OptionSetting> optionSettings) {
+        return Optional.ofNullable(optionSettings).orElse(Collections.emptyList())
+                .stream()
+                .map(optionSetting -> {
+                    return OptionSetting.builder()
+                            .name(optionSetting.name())
```

Comment:
Could elaborate why we didn't include `allowedValues, dataType, defaultValue, isModifiable` in our schema and mapping?

Reply:
Backwards compatibility mainly: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-optiongroup-optionconfigurations-optionsettings.html
These could be implemented afterwards if needed. This PR is not aiming to expand the existing functionality.